pkg/api/defaults.go

```diff
@@ -438,19 +438,29 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 			profile.AcceleratedNetworkingEnabledWindows = helpers.PointerToBool(DefaultAcceleratedNetworkingWindowsEnabled)
 		}
 
-		if profile.Distro == "" && profile.OSType != Windows {
-			if p.OrchestratorProfile.IsKubernetes() {
-				if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < VHDDiskSizeAKS {
-					profile.Distro = Ubuntu
-				} else {
-					if IsNSeriesSKU(p) {
-						profile.Distro = AKSDockerEngine
+		if profile.OSType != Windows {
+			if profile.Distro == "" {
+				if p.OrchestratorProfile.IsKubernetes() {
+					if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < VHDDiskSizeAKS {
+						profile.Distro = Ubuntu
 					} else {
-						profile.Distro = AKS
+						// Ensure distro is set properly for N Series SKUs, because
+						// (1) At present, "aks-docker-engine" and "ubuntu" are the only working distro base for running GPU workloads on N Series SKUs
+						// (2) Previous versions of acs-engine had working implementations using the "aks" distro value,
+						//     so we need to hard override it in order to produce a working cluster in upgrade/scale contexts
+						if profile.IsNSeriesSKU() {
+							profile.Distro = AKSDockerEngine
+						} else {
+							profile.Distro = AKS
+						}
 					}
+				} else if !p.OrchestratorProfile.IsOpenShift() {
+					profile.Distro = Ubuntu
+				}
+			} else if p.OrchestratorProfile.IsKubernetes() && (isUpgrade || isScale) && profile.IsNSeriesSKU() {
+				if profile.Distro != Ubuntu {
```

Comment:
I have a slight preference for `if profile.Distro == AKS` for now in case we add another distro in the future that we don't want to override in upgrade/scale